src/sagas/nanoContract.js

```diff
@@ -311,11 +317,43 @@ export function* unregisterNanoContract({ payload }) {
   yield put(nanoContractUnregisterSuccess({ ncId }));
 }
 
+/**
+ * Process update on registered Nano Contract address to persist on store.
+ * @param {{
+ *   payload: {
+ *     ncId: string;
+ *     address: string;
+ *   }
+ * }}
+ */
+export function* requestNanoContractAddressChange({ payload }) {
+  const { ncId, address } = payload;
+
+  const wallet = yield select((state) => state.wallet);
+  if (!wallet.isReady()) {
+    log.error('Fail updating Nano Contract address because wallet is not ready yet.');
+    // This will show user an error modal with the option to send the error to sentry.
+    yield put(onExceptionCaptured(new Error(failureMessage.walletNotReadyError), true));
+    return;
+  }
+
+  yield call(
+    [
+      wallet.storage,
+      wallet.storage.updateNanoContractRegisteredAddress
```

Comment:
I think you need to create this method also in the `src/store.js` file, don't you?